mountpoint-s3-client/src/object_client.rs

```diff
@@ -20,11 +22,22 @@ impl ETag {
         &self.etag
     }
 
+    // Creating default etag for tests
     pub fn for_tests() -> Self {
         Self {
             etag: "test_etag".to_string(),
         }
     }
+
+    // Creating unique etag from bytes
+    pub fn from_object_bytes(data: &[u8]) -> Self {
+        let mut hasher = Md5::new();
+        hasher.update(data);
+
+        let hash = hasher.finalize();
```

Comment:
nit:  have you considered something like this:
```suggestion
        let hash = Md5::new()
            .update(data)
            .finalize();
```
may be subjective though :)